internal/store/service.go

```diff
@@ -184,6 +185,49 @@ func serviceMetricFamilies(allowAnnotationsList, allowLabelsList []string) []gen
 				}
 			}),
 		),
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_service_ports",
+			"Metric providing details about the ports exposed by services.",
+			metric.Gauge,
+			basemetrics.ALPHA,
+			"",
+			wrapSvcFunc(func(e *v1.Service) *metric.Family {
+				ms := []*metric.Metric{}
+				labelKeys := []string{"port_protocol", "port_number", "port_name", "port_target", "port_node_number", "port_app_protocol"}
```

Comment:
It has the cardinality issue.

Reply:
@CatherineF-dev we've the same requirement and would appreciate if this could be merged.

Reply:
@CatherineF-dev any update on this?

Reply:
I agree with @CatherineF-dev this metrics will generate too many timeseries. Because of that I should we should make it opt-in. You can do that by using the `generator.NewOptInFamilyGenerator`.